src/java/org/apache/cassandra/io/util/ChecksumWriter.java

```diff
@@ -91,11 +91,38 @@ public void appendDirect(ByteBuffer bb, boolean checksumIncrementalResult)
     }
 
     public void writeFullChecksum(@Nonnull File digestFile)
+    {
+        writeFullChecksum(digestFile, fullChecksum.getValue());
+    }
+
+    /**
+     * Repopulate checksum value from given source file and write to digest file
+     */
+    public void repopulateChecksum(@Nonnull File source, @Nonnull File digestFile)
+    {
+        try (RandomAccessReader in = RandomAccessReader.open(source))
+        {
+            CRC32 checksum = new CRC32();
+            CheckedInputStream checkedInputStream = new CheckedInputStream(in, checksum);
```

Comment:
Are we sure this recompute the checksum the same way than when we don't recompute it?

To be fair, I'm not that familiar with these area of the code, but scrub calls `appendDirect` with `checksumIncrementalResult = true`, and that means each "buffer" individual CRC is added to the full CRC, but afaict, `CheckedInputStream` will not do that.

And more generally, `ChecksumWriter` has kind of 2 "mode" due to this ``checksumIncrementalResult` flag, that produce 2 different CRC, so adding a `repopulateChecksum` method that does not distinguish between those 2 mode feels a big dodgy in the first place.

Reply:
> Are we sure this recompute the checksum the same way than when we don't recompute it?

The compressed writer writes a chunk of compressed buffer and then writes the checksum of this compressed buffer into data channel.  So `ChecksumWriter#fullChecksum` updates the compressed buffer and then updates the checksum of this buffer.

Iterating data file will generate the same checksum as above.

Reply:
> And more generally, ChecksumWriter has kind of 2 "mode" due to this ``checksumIncrementalResultflag, that produce 2 different CRC, so adding arepopulateChecksum` method that does not distinguish between those 2 mode feels a big dodgy in the first place.

the other mode (checksumIncrementalResultflag =false) is not used by compressed writer..

Reply:
> The compressed writer writes a chunk of compressed buffer and then writes the checksum of this compressed buffer into data channel.

Hum, ok. I was kind of looking for that but couldn't find where that happened. I can see this now this happens through the `incrementalOut` of `ChecksumWriter`, which I didn't quite follow initially.

> the other mode (checksumIncrementalResultflag =false) is not used by compressed writer.

Yes, but the method here is added to `ChecksumWriter`, not the compressed writer. I'm not saying it's incorrect, I'm saying it's dodgy because this make it easy to re-use this new method in other use of `ChecksumWriter` without realizing it only work in specific circumstance. Tl;dr, I'm ok not adding complexity to the method, but let's at least add a clear comment on the precondition necessary to using that method correctly.

Reply:
I moved the data file scanning into `CompressedSequentialWriter` in https://github.com/datastax/cassandra/pull/1197/commits/de441b5f7eff5032128dd79ee4f7cd038d3fc006 and added a comment in `ChecksumWriter`